### PR TITLE
Add ecr scan step

### PR DESCRIPTION
**Pull request by danstn**
This will introduce a soft-fail step that will add ECR scan results to build annotation.

### CODE REVIEW

**Comment by dannymidnight** [comment]
Nice one. Thanks @danstn

**Review by dannymidnight — APPROVED** [review]

File: .buildkite/pipeline.deploy.yml [hunk 1]
```diff
@@ -12,6 +12,13 @@ steps:
         login: true
         account-ids: ${ECR_ACCOUNT_ID}
 
+  - name: ":ecr: ECR Vulnerabilities Scan"
+    command: "true"
+    soft_fail: true
+    plugins:
```
Comment (yob) [inline .buildkite/pipeline.deploy.yml:18]: does this need a `depends_on` to ensure it runs after the image is pushed?

File: .buildkite/pipeline.deploy.yml [hunk 2]
```diff
@@ -12,6 +12,13 @@ steps:
...
+      - buildkite/ecr-scan-results#v1.2.0:
+          image-name: "${ECR_REPO}:${BUILDKITE_BUILD_NUMBER}"
+
   # If the current user is part of the deploy team, then wait for everything to
   # finish before deploying
   - wait: ~
```
Comment (yob) [inline .buildkite/pipeline.deploy.yml:24]: does this wait need to depend on the ECR scan? Maybe not now I guess, while the ECR scan is soft fail?

File: .buildkite/pipeline.deploy.yml (hunk 1 above)
Comment (danstn) [reply .buildkite/pipeline.deploy.yml:19]: Yes. Updated here https://github.com/buildkite/docs/pull/2519/commits/6ce10328bfc0350cd5cf49d91687ecf85aedeafd.

**Pull request merged**